File: packages/android/src/main/java/com/kozmos/components/Listbox/Listbox.figma.kt
package com.kozmos.components.listbox

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=401-9475")
class KozmosListboxConnect {
    @FigmaProperty(FigmaType.Enum, "Selection")
    val multiple: Boolean = Figma.mapping(
        "Single" to false,
        "Multiple" to true
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val disabled: Boolean = Figma.mapping(
        "Default" to false,
        "Focus" to false,
        "Disabled" to true
    )

    @FigmaProperty(FigmaType.Text, "Option 1 Text")
    val option1: String = "Metro Station"

    @FigmaProperty(FigmaType.Text, "Option 1 Description")
    val option1Description: String = "Fast transit connection"

    @FigmaProperty(FigmaType.Text, "Option 2 Text")
    val option2: String = "Bus Stop"

    @FigmaProperty(FigmaType.Text, "Option 2 Description")
    val option2Description: String = "Frequent local service"

    @FigmaProperty(FigmaType.Text, "Option 3 Text")
    val option3: String = "Bike Parking"

    @FigmaProperty(FigmaType.Text, "Option 3 Description")
    val option3Description: String = "Secure racks nearby"

    @Composable
    fun ComponentExample() {
        KozmosListbox(
            options = listOf(
                KozmosListboxOption("metro-station", option1, option1Description),
                KozmosListboxOption("bus-stop", option2, option2Description),
                KozmosListboxOption("bike-parking", option3, option3Description)
            ),
            selectedValues = if (multiple) listOf("metro-station", "bus-stop") else listOf("metro-station"),
            multiple = multiple,
            enabled = !disabled
        )
    }
}

File: packages/ios/Sources/Components/Listbox/Listbox.figma.swift
import SwiftUI
import Figma

struct KozmosListboxConnect: FigmaConnect {
    let component = KozmosListbox.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=401-9475"

    @FigmaEnum(
        "Selection",
        mapping: [
            "Single": false,
            "Multiple": true
        ]
    )
    var multiple: Bool = false

    @FigmaEnum(
        "State",
        mapping: [
            "Default": false,
            "Focus": false,
            "Disabled": true
        ]
    )
    var disabled: Bool = false

    @FigmaString("Option 1 Text")
    var option1: String = "Metro Station"

    @FigmaString("Option 1 Description")
    var option1Description: String = "Fast transit connection"

    @FigmaString("Option 2 Text")
    var option2: String = "Bus Stop"

    @FigmaString("Option 2 Description")
    var option2Description: String = "Frequent local service"

    @FigmaString("Option 3 Text")
    var option3: String = "Bike Parking"

    @FigmaString("Option 3 Description")
    var option3Description: String = "Secure racks nearby"

    var body: some View {
        KozmosListbox(
            options: [
                KozmosListboxOption(value: "metro-station", label: self.option1, description: self.option1Description),
                KozmosListboxOption(value: "bus-stop", label: self.option2, description: self.option2Description),
                KozmosListboxOption(value: "bike-parking", label: self.option3, description: self.option3Description)
            ],
            selectedValues: .constant(self.multiple ? ["metro-station", "bus-stop"] : ["metro-station"]),
            multiple: self.multiple,
            disabled: self.disabled
        )
    }
}
